### PR TITLE
fix: improve error hints when select * from xx.yy

### PR DESCRIPTION
**Pull request by ZhengLin-Li**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

- Fixes https://github.com/datafuselabs/databend/issues/13718

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14663)

### CODE REVIEW

**Comment by sundy-li** [comment]
Can you add   logic tests to cover the issue case ?

**Comment by ZhengLin-Li** [comment]
@sundy-li Hi，thanks for your comments. Could I ask where can i add such test?

**Comment by BohuTANG** [comment]
You can add a new test like `issue_13718.test` in https://github.com/datafuselabs/databend/tree/main/tests/sqllogictests/suites/base/issues.

File: tests/sqllogictests/suites/base/issues/issue_13718.test [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+query A
+select * from xx.yy;
+----
+1025: error:
+  --> SQL:3:15
+  |
+1 |
+2 |
+3 | select * from xx.yy;
```
Comment (BohuTANG) [inline tests/sqllogictests/suites/base/issues/issue_13718.test:12]:
```suggestion
statement error (?s)1003.*Unknown database `xx` in catalog 'default'
select * from xx.yy;

```

File: tests/sqllogictests/suites/base/issues/issue_13718.test [hunk 2]
```diff
@@ -0,0 +1,7 @@
+statement ok
+DROP DATABASE IF EXISTS xx
+
+statement error (?s)1003.*Unknown database `xx` in catalog 'default'
+select * from xx.yy;
+
+  |               ^^ Unknown database `xx` in catalog 'default'
```
Comment (BohuTANG) [inline tests/sqllogictests/suites/base/issues/issue_13718.test:7]: This line is no need.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**